File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        l,r,total=0,0,0
        c=0
        while r<len(nums):
            total+=nums[r]
            while nums[r]*(r-l+1)>total+k:
                total-=nums[l]
                l+=1
            c=max(c,r-l+1)
            r+=1
        return c
        
           

File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        long long l=0, r=-1, total=0, cnt=0;
        while(++r<nums.size()){
            total+=nums[r];
            while(nums[r]*(r-l+1)>total+k)  total-=nums[l++];
            cnt = max(r-l+1, cnt);
        }
        return (int)cnt;
    }
};